examples/solver/dumpsad-solver/src/astromesh.rs

```diff
@@ -0,0 +1,150 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Binary, Coin, DenomMetadata, Int128, Uint128, Uint64};
+use sha2::{Digest, Sha256};
+use tiny_keccak::{Hasher, Keccak};
+
+pub const PLANE_COSMOS: &str = "COSMOS";
+
+pub const ACTION_COSMOS_INVOKE: &str = "COSMOS_INVOKE";
+
+pub const QUERY_ACTION_COSMOS_QUERY: &str = "COSMOS_QUERY";
+pub const QUERY_ACTION_COSMOS_BANK_BALANCE: &str = "COSMOS_BANK_BALANCE";
+pub const QUERY_ACTION_COSMOS_KVSTORE: &str = "COSMOS_KVSTORE";
+
+#[cw_serde]
+pub struct MsgAstroTransfer {
+    #[serde(rename = "@type")]
+    pub ty: String,
+    pub sender: String,
+    pub receiver: String,
+    pub src_plane: String,
+    pub dst_plane: String,
+    pub coin: Coin,
+}
+
+impl MsgAstroTransfer {
+    pub fn new(
+        sender: String,
+        receiver: String,
+        src_plane: String,
+        dst_plane: String,
+        coin: Coin,
+    ) -> Self {
+        MsgAstroTransfer {
+            ty: "/flux.astromesh.v1beta1.MsgAstroTransfer".to_string(),
+            sender,
+            receiver,
+            src_plane,
+            dst_plane,
+            coin,
+        }
+    }
+}
+
+#[cw_serde]
+pub struct InitialMint {
```

Comment:
The token is minted as people purchase the token, not upfront.

So this can be problem with naming or logic, we can sync more.

Reply:
> So this is to create 1 token so we have the denom in bank?
Yes

> The token is minted as people purchase the token, not upfront.

I kinda simulate pump fun when it's minted 10^9 token initially then distribute them as people buy